fitbenchmarking/fitbenchmarking
# Issue #322: Refactor functions var in fitting_problem

#### Description of Work

Fixes #322 
Swaps from using a list of (callable, parameter) pairs to having a single callable, and changes how parameters are stored by the fitting problem.

This allows removing the function_id from calls to eval_f.

#### Testing Instructions

1. Verify accuracy does not change between master and branch. (I checked this already but it's good to be sure)


Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
